Makefile

```diff
@@ -300,16 +295,8 @@ $(UI_DIR)/node_modules:
 gen-ui: $(UI_DIR)/node_modules  ## Build UI web app
 	cd $(UI_DIR) && $(NPM) run build
 
-proto: tools ## Build proto (Protocol Buffers) files
-	$(PROTOC) --proto_path=pkg/actions --go_out=pkg/actions --go_opt=paths=source_relative actions.proto
-	$(PROTOC) --proto_path=pkg/auth/model --go_out=pkg/auth/model --go_opt=paths=source_relative model.proto
-	$(PROTOC) --proto_path=pkg/catalog --go_out=pkg/catalog --go_opt=paths=source_relative catalog.proto
-	$(PROTOC) --proto_path=pkg/gateway/multipart --go_out=pkg/gateway/multipart --go_opt=paths=source_relative multipart.proto
-	$(PROTOC) --proto_path=pkg/graveler --go_out=pkg/graveler --go_opt=paths=source_relative graveler.proto
-	$(PROTOC) --proto_path=pkg/graveler/committed --go_out=pkg/graveler/committed --go_opt=paths=source_relative committed.proto
-	$(PROTOC) --proto_path=pkg/graveler/settings --go_out=pkg/graveler/settings --go_opt=paths=source_relative test_settings.proto
-	$(PROTOC) --proto_path=pkg/kv --go_out=pkg/kv --go_opt=paths=source_relative secondary_index.proto
-	$(PROTOC) --proto_path=pkg/kv/kvtest --go_out=pkg/kv/kvtest --go_opt=paths=source_relative test_model.proto
+gen-proto: ## Build Protocol Buffers (proto) files using Buf CLI
+	buf generate
```

Comment:
So now users who want to develop lakeFS need to install the buf CLI on their env?
Previously we used a docker so no added installation was needed.
I think this needs to be reflected in our contributing guide (it seems that this step is optional)

Reply:
We commit the source code, so usually contribute will not require to build proto.
Also, protoc doesn't have a formal docker image and we build it and need to maintain it.
I've added a note to the contrib guide to install the cli the case proto files require an update.

Reply:
@N-o-Z updated the PR - used `go run` to run buf cli, so now it is no diff from running lint or other Go dependencies.